packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-processing-job.ts

```diff
@@ -0,0 +1,438 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Duration, Lazy, Size, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs'; import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { AppSpecification, ExperimentConfig, NetworkConfig, ProcessingInput, ProcessingOutputConfig, ProcessingResources, StoppingCondition } from './base-types';
+import { renderTags } from './private/utils';
+
+/**
+ * Properties for creating an Amazon SageMaker processing job
+ *
+ * @experimental
+ */
+export interface SageMakerCreateProcessingJobProps extends sfn.TaskStateBaseProps {
+  /**
+   * Role for the Processing Job. The role must be granted all necessary permissions for the SageMaker processing job to
+   * be able to operate.
+   *
+   * See https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-createprocessingjob-perms
+   *
+   * @default - a role with appropriate permissions will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Configures the processing job to run a specified Docker container image.
+   */
+  readonly appSpecification: AppSpecification;
+
+  /**
+   * Sets the environment variables in the Docker container.
+   *
+   * @default - No environment variables
+   */
+  readonly environment?: { [key: string]: any };
+
+  /**
+   * Associates a SageMaker job as a trial component with an experiment and trial. Specified when you call the
+   * following APIs: CreateProcessingJob, CreateTrainingJob, CreateTransformJob
+   *
+   * @default - No experiment configuration
+   */
+  readonly experimentConfig?: ExperimentConfig;
+
+  /**
+   * Networking options for a processing job.
+   *
+   * @default - No network configuration
+   */
+  readonly networkConfig?: NetworkConfig;
+
+  /**
+   * For each input, data is downloaded from S3 into the processing container before the processing job begins running
+   * if 'S3InputMode' is set to File.
+   *
+   * @default - No processing inputs
+   */
+  readonly processingInputs?: ProcessingInput[];
+
+  /**
+   * The name of the processing job. The name must be unique within an AWS Region in the AWS account.
+   */
+  readonly processingJobName: string;
```

Comment:
I think this may definitely be a good idea for other aws-stepfunctions-tasks constructs, it might not apply so nicely to SageMaker tasks.

If I supplied a default name, it would only work the very first time the task is created & used in a step functions workflow since the job name has to be unique. 

Consider a simple scenario:
* Someone creates a Step Functions workflow in CDK
* The workflow creates (this) SageMaker Processing Job Task
* The Step Functions Workflow is run, creating an actual SageMaker processing Job.
* The (default) name is used. The job is invoked as expected.
* The Step Functions Workflow is run a second time
* We try to use the (default) name again. We fail because it isn't unique

Even if we tried to use a UUID as a default name here, the job name would only be different for the extent of that CDK deployment. Practically, what needs to happen here is that the job name needs to be generated and supplied during runtime.

One solution could be to default to using the step functions execution ID. However, the length of that ID can be longer than a SageMaker processing Job name. So I think it might be better to have no default here (rather than resorting to strange default behavior that might confuse a customer).